### DISCUSSION TITLE
Imgur

### DISCUSSION THREAD

**Discussion by SefaAcarsoy** [flameshot-org/flameshot#3789] [General]
Hello. Imgur is blocked in Turkiye.
That's why uploaded images cannot be viewed.
I wonder if the possibility to upload to a provider other than imgur will come?

**Comment by mmahmoudian** [comment]
Hi, at the moment it is not possible, but there are feature requests available to add other platforms. It was decided by the core team that this feature should only be available via a plugin system (for which we already have a prototype Pull Request).

In the meanwhile, you can use the `--raw` argument to get the screenshot and pass that to any other software or script that can upload the picture to your preferred image hosting platform.

**Comment by SefaAcarsoy** [comment]
How is it possible? O mean --raw argument etc?
________________________________
Gönderen: Mehrad Mahmoudian ***@***.***>
Gönderildi: 30 Kasım 2024 Cumartesi 18:46
Kime: flameshot-org/flameshot ***@***.***>
Bilgi: SefaAcarsoy ***@***.***>; Author ***@***.***>
Konu: Re: [flameshot-org/flameshot] Imgur (Discussion #3789)


Hi, at the moment it is not possible, but there are feature requests available to add other platforms. It was decided by the core team that this feature should only be available via a plugin system (for which we already have a prototype Pull Request).

In the meanwhile, you can use the --raw argument to get the screenshot and pass that to any other software or script that can upload the picture to your preferred image hosting platform.

**Reply by mmahmoudian** [reply]
I assume you use Mac or Linux, as Flameshot doesn't have a command line interface for Windows users.

If you have a script to upload to something similar to imgur, you can do this:

```sh
flameshot gui --raw | my_upload_script.sh`
```

Alternatively you can save it to a file and then run a script that uploads the file:

```sh
TEMP_FILE="$(mktemp)" && flameshot gui --path "${TEMP_FILE}" && my_upload_script.sh "${TEMP_FILE}"
```

**Comment by SefaAcarsoy** [comment]
Hello,

Unfortunately, Windows 🙂

________________________________
Gönderen: Mehrad Mahmoudian ***@***.***>
Gönderildi: 22 Şubat 2025 Cumartesi 09:47
Kime: flameshot-org/flameshot ***@***.***>
Bilgi: SefaAcarsoy ***@***.***>; Author ***@***.***>
Konu: Re: [flameshot-org/flameshot] Imgur (Discussion #3789)


I assume you use Mac or Linux, as Flameshot doesn't have a command line interface for Windows users.

If you have a script to upload to something similar to imgur, you can do this:

flameshot gui --raw | my_upload_script.sh`

Alternatively you can save it to a file and then run a script that uploads the file:

TEMP_FILE="$(mktemp)" && flameshot gui --path "${TEMP_FILE}" && my_upload_script.sh "${TEMP_FILE}"

**Reply by mmahmoudian** [reply]
I'm sorry, we have not implemented CLI for Windows yet. It is in the ToDo list though.
